### ISSUE TITLE
Various strings for some dialogs don't get translated properly in the editor

### ISSUE THREAD

**Issue by foxydevloper** [godotengine/godot#50650]
### Godot version

v4.0.dev.custom_build [bb037de95]

### System information

Windows 10

### Issue description

It seems that https://github.com/godotengine/godot/pull/46735 broke translations for some dialogs' buttons inside of the editor. They used to be properly translated in the editor in 3.x, but now "Cancel", "OK", etc doesn't get translated properly in the editor. (This image is the editor in Spanish. Notice the "Cancel", which there is a Spanish translation entry for)
![image](https://user-images.githubusercontent.com/12120644/126354344-0b329f2f-774c-4608-a5ae-8124c5b3cd78.png)

It seems like it's all of these strings that are broken that use TTRC instead of TTR or RTR:
![image](https://user-images.githubusercontent.com/12120644/126353505-4cbe34d3-a606-4c3b-89f4-ab7105e8a1ab.png)

I don't think it's as simple as converting from TTRC to RTR since that would break what #46735 tried to fix , but it is a regression from 3.x

### Steps to reproduce

Look at any dialog with "Cancel", "OK", etc

**Comment by bruvzg** [comment]
For the reference: `TTRC` macro do absolutely nothing, its only purpose is adding strings to the POT file (it's parsed by `extract.py`).

Translation macros are:
- `TTR/TTRN` - for editor only translations, string is added to POT.
- `RTR/RTRN` - for both editor and runtime translation, string is added to POT.
- `DTR/DTRN` - for documentation, uses different translation source, string is NOT added to POT (which is generated from XML documentation source instead).
- `TTRGET` - same as `TTR` but string is NOT added to POT.

The only valid case for `TTRC` seems to be in the combination with `TTRGEN`:
```c++
const char *something[] = { TTRC("a"), TTRC("b"), TTRC("c") }; // Define strings and add them to POT

...

translated = TTRGET(something[i]); // Translate without adding "something[i]" to POT.
```

**Comment by KoBeWi** [comment]
The dialogs were previously using RTRs, but it was causing issues with some properties. E.g. `title` property was translated and the translated string was saved in the scene. It was replaced with TTRC. It still works at runtime, due to auto-translation of strings, but seems like editor strings aren't translated. Not sure how it could be solved, other than enabling auto-translation in editor (at least for some nodes) or explicitly translating all dialogs.

**Comment by KoBeWi** [comment]
Ok easy fix is to add a new method that translates in the editor, but does nothing at runtime:
```C++
String TTR2(const String &p_text, const String &p_context) {
#ifdef TOOLS_ENABLED
	return TTR(p_text, p_context);
#else
	return p_text;
#endif
}
```
Also none of the properties can be affected by this macro, so accessors like this
https://github.com/godotengine/godot/blob/e930c8d3838280e40baabc4426bd8236f7ac50a3/scene/gui/dialogs.cpp#L425-L431
need to be reworked by adding an actual property with untranslated text.

**Comment by KoBeWi** [comment]
Looks like something broke with `tr()`. It doesn't translate strings in editor, while `TTR()` does. That PR didn't touch it though...

**Comment by akien-mga** [comment]
For the record, next time I would suggest opening a new bug report even if the symptoms are similar. This bug was fixed in 4.1 already, and re-closing it now as fixed in 4.3 doesn't make much sense.
